app/email_branding/rest.py

```diff
@@ -18,6 +19,21 @@
 register_errors(email_branding_blueprint)
 
 
+@email_branding_blueprint.errorhandler(IntegrityError)
+def handle_integrity_error(exc):
+    """
+    Handle integrity errors caused by the unique constraint
+    """
+    if 'uq_email_branding_name' in str(exc):
+        return jsonify(
+            result='error',
+            message={'name': ["An email branding with that name already exists."]}
+        ), 400
+
+    current_app.logger.exception(exc)
+    return jsonify(result='error', message="Internal server error"), 500
```

Comment:
Wouldn't it be better to pass the error message we get here? "Internal server error" doesn't tell us much.

Reply:
It might be, but I'm not here to fix every problem 😅  This is exactly what the letter branding integrity exception handler does - and it's not worse than existing behaviour which would just bubble up the 500.

Just looking for a small+quick incremental improvement here (since it's unplanned work) :)

Reply:
ok, fair do